### PR TITLE
Allow Integration Script to Output Markdown

### PR DESCRIPTION
**Pull request by NickGerleman**
This adds a flag to the RN integration script to generate a markdown report with status of checks. This will be used for automated integration as the body of pull requests created by the bot.

We additionally automate the addition of a URL in reports to show new commits since last integration.

Added some simple UTs for the logger and manually validated against an integration.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/react-native-windows/pull/6862)

### CODE REVIEW

**Review by NikoAri — APPROVED** [review]
:shipit:

File: packages/@rnw-scripts/integrate-rn/src/test/logger.test.ts [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+× yarn build
+`,
+  );
+});
+
+function stripAnsiColors(str: string) {
+  // Taken from https://github.com/chalk/ansi-regex/blob/master/index.js
+  const pattern = [
+    '[\\u001B\\u009B][[\\]()#;?]*(?:(?:(?:[a-zA-Z\\d]*(?:;[-a-zA-Z\\d\\/#&.:=?%@~_]*)*)?\\u0007)',
+    '(?:(?:\\d{1,4}(?:;\\d{0,4})*)?[\\dA-PR-TZcf-ntqry=><~]))',
```
Comment (NikoAri) [inline packages/@rnw-scripts/integrate-rn/src/test/logger.test.ts:80]: Isn't there an existing const like this somewhere?

Comment (NickGerleman) [reply packages/@rnw-scripts/integrate-rn/src/test/logger.test.ts:80]: Nothing I could find without pulling in a separate package.

**Review by dannyvv — APPROVED** [review]

**Comment by NickGerleman** [comment]
Going to merge this one in before the 10 hours because I'm experimenting with a pipeline dependent on this.

**Pull request merged**